### PR TITLE
add 'skip_on_windows()' (closes #289)

### LINKED ISSUE #289: test-watcher hangs when run on Windows
I'm guessing the use of shell utilities doesn't work on Windows.

### PR DESCRIPTION
**Pull request by kevinushey**
This PR adds `skip_on_windows()` and uses it on a `watcher` test that otherwise hangs.

### CODE REVIEW

**Comment by hadley** [comment]
And could you please merge/rebase?

**Comment by kevinushey** [comment]
Rebased!

**Comment by kevinushey** [comment]
Okay, done now (with chains).

**Comment by kevinushey** [comment]
Using `skip_on_os` now -- let me know what you think.

File: R/utils.r [hunk 1]
```diff
@@ -65,6 +65,43 @@ skip_on_cran <- function() {
...
+    skip("On Linux")
+  else if (sysname == "sunos" && "solaris" %in% systems)
+    skip("On Solaris")
+
+  invisible(TRUE)
+}
+
+#' @export
+#' @rdname skip
+skip_on_windows <- function() skip_on_os("windows")
```
Comment (hadley) [inline R/utils.r]: I'm not sure if the helpers are necessary - they don't save that much typing or add much extra clarity. But I can remove them.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**